aten/src/ATen/functorch/BatchRulesLoss.cpp

```diff
@@ -46,6 +46,27 @@ mse_loss_batch_rule(const at::Tensor& self, optional<int64_t> self_bdim, const a
   TORCH_INTERNAL_ASSERT(false);
 };
 
+std::tuple<at::Tensor,optional<int64_t>>
+smooth_l1_loss_batch_rule(const at::Tensor& self, optional<int64_t> self_bdim, const at::Tensor& target,
```

Comment:
I think it would be nice if we can refactor the code so that we have a same helper function/template for `mse_loss` and `smooth_l1_loss` (as all of the code is same except the call to respective function).

Reply:
I agree, I will refactor it once I figure out why some of the tests failed in the latest commit. (Please also let me know if you know what the error is)